tsl/test/expected/dist_hypertable-14.out

```diff
@@ -5020,6 +5020,70 @@ SELECT count(*) FROM disttable;
     16
 (1 row)
 
+-- Show that DataNodeCopy is NOT used when source hypertable and target hypertable
+-- of the SELECT are both distributed
+EXPLAIN (COSTS OFF)
+INSERT INTO disttable (time, device, temp_c)
+SELECT time, device, temp_c FROM disttable;
+                                    QUERY PLAN                                     
+-----------------------------------------------------------------------------------
+ Custom Scan (HypertableInsert)
+ Insert on distributed hypertable disttable
+   ->  Insert on disttable
+         ->  Custom Scan (DataNodeDispatch)
+               Batch size: 1000
+               ->  Custom Scan (ChunkDispatch)
+                     ->  Append
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_2
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_3
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_4
+(10 rows)
+
+EXPLAIN (COSTS OFF)
+INSERT INTO disttable (time, device, temp_c)
+SELECT time, device, temp_c FROM disttable LIMIT 1;
+                                             QUERY PLAN                                              
+-----------------------------------------------------------------------------------------------------
+ Custom Scan (HypertableInsert)
+ Insert on distributed hypertable disttable
+   ->  Insert on disttable
+         ->  Custom Scan (DataNodeDispatch)
+               Batch size: 1000
+               ->  Custom Scan (ChunkDispatch)
+                     ->  Subquery Scan on "*SELECT*"
+                           ->  Limit
+                                 ->  Custom Scan (AsyncAppend)
+                                       ->  Append
+                                             ->  Custom Scan (DataNodeScan) on disttable disttable_2
+                                             ->  Custom Scan (DataNodeScan) on disttable disttable_3
+                                             ->  Custom Scan (DataNodeScan) on disttable disttable_4
+(13 rows)
+
+EXPLAIN (COSTS OFF)
+INSERT INTO disttable (time, device, temp_c)
+SELECT time, device, temp_c FROM disttable RETURNING *;
+                                    QUERY PLAN                                     
+-----------------------------------------------------------------------------------
+ Custom Scan (HypertableInsert)
+ Insert on distributed hypertable disttable
+   ->  Insert on disttable
+         ->  Custom Scan (DataNodeDispatch)
+               Batch size: 1000
+               ->  Custom Scan (ChunkDispatch)
+                     ->  Append
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_2
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_3
+                           ->  Custom Scan (DataNodeScan) on disttable disttable_4
+(10 rows)
+
+INSERT INTO disttable (time, device, temp_c)
```

Comment:
Could you also add a test for this case?
INSERT INTO dist2 SELECT  * FROM dist_foo
a) source and target are different  and b) uses *

Reply:
Done. Added a test case for this.